src/sinks/websocket/sink.rs

```diff
@@ -267,11 +272,14 @@ impl WebSocketSink {
                 },
 
                 event = input.next() => {
-                    if event.is_none() {
+                    let mut event = if let Some(event) = event {
+                        event
+                    } else {
                         break;
-                    }
-                    let log = encode_event(event.unwrap(), &self.encoding);
-                    let res = match log {
+                    };
+                    let mut bytes = BytesMut::new();
+                    self.transformer.transform(&mut event);
+                    let res = match self.encoder.encode(event, &mut bytes).ok().map(|_| Message::text(String::from_utf8_lossy(&bytes))) {
```

Comment:
`encode()` is responsible for generating the internal logs and metrics for failed encodings, right?

Reply:
Added in c8136a0a747ab401bdf35a84f24bafd3ec5ab4ee